compiler/test/dotty/tools/repl/DocTests.scala

```diff
@@ -0,0 +1,209 @@
+package dotty.tools
+package repl
+
+import org.junit.Test
+import org.junit.Assert.assertEquals
+
+class DocTests extends ReplTest {
+  @Test def docOfDef =
+    fromInitialState { implicit s => run("/** doc */ def foo = 0") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfVal =
+    fromInitialState { implicit s => run("/** doc */ val foo = 0") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfObject =
+    fromInitialState { implicit s => run("/** doc */ object Foo") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfClass =
+    fromInitialState { implicit s => run("/** doc */ class Foo") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc new Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfTrait =
+    fromInitialState { implicit s => run("/** doc */ trait Foo") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc new Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfDefInObject =
+    fromInitialState { implicit s => run("object O { /** doc */ def foo = 0 }") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc O.foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfValInObject =
+    fromInitialState { implicit s => run("object O { /** doc */ val foo = 0 }") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc O.foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfObjectInObject =
+    fromInitialState { implicit s => run("object O { /** doc */ object Foo }") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc O.Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfClassInObject =
+    fromInitialState { implicit s => run("object O { /** doc */ class Foo }") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc new O.Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfTraitInObject =
+    fromInitialState { implicit s => run("object O { /** doc */ trait Foo }") }
+    .andThen { implicit s =>
+      storedOutput()
+      run(":doc new O.Foo")
+      assertEquals("/** doc */", storedOutput().trim)
+    }
+
+  @Test def docOfDetInClass =
+    fromInitialState { implicit s => run("class C { /** doc */ def foo = 0 }") }
+    .andThen { implicit s => run("val c = new C") }
```

Comment:
There is no need for separate runs. You can have muti-line input:
```scala
val source =
  """class C { /** doc */ def foo = 0 }
    |
    |val c = new C""".stripMargin
run(source)
```
There is value in using multiple run if you want to overwrite a definition. E.g:
```scala
scala> /** doc 1 */ class Foo
// defined class Foo

scala> :doc new Foo
/**  doc 1 */

scala> /** doc 2 */ class Foo
// defined class Foo

scala> :doc new Foo
/**  doc 2 */
```